corehq/apps/settings/views.py

```diff
@@ -88,12 +90,19 @@ def get(self, request, *args, **kwargs):
 class MyAccountSettingsView(BaseMyAccountView):
     urlname = 'my_account_settings'
     page_title = ugettext_lazy("My Information")
+    api_key = None
 
     @method_decorator(check_preview_bootstrap3())
     @method_decorator(use_select2())
     def dispatch(self, request, *args, **kwargs):
         return super(MyAccountSettingsView, self).dispatch(request, *args, **kwargs)
 
+    def get_or_create_api_key(self):
+        if not self.api_key:
+            api_key, _ = ApiKey.objects.get_or_create(user=self.request.user)
```

Comment:
Unfortunately, the django get_or_create isn't guaranteed to be atomic, so to bulletproof this kind of thing I like to wrap these kinds of calls in a `CriticalSection`. So for example wrapping it in something like: `with CriticalSection(['get-or-create-api-key-for-<userid>']):`.  This way, in case the page ends up loading twice simultaneously because of someone accidentally double clicking, we don't generate two api keys.